cpp/memilio/utils/stl_util.h

```diff
@@ -65,23 +65,24 @@ inline std::ostream& set_ostream_format(std::ostream& out, size_t width, size_t
  * @return iterator to inserted or replaced item in vec
  */
 template <typename T, typename Pred>
-typename std::vector<T>::iterator insert_sorted_replace(std::vector<T>& vec, T const& item, Pred pred)
+void insert_sorted_replace(std::vector<T>& vec, T const& item, Pred pred)
 {
     auto bounds = std::equal_range(begin(vec), end(vec), item, pred);
     auto lb     = bounds.first;
     auto ub     = bounds.second;
     assert(ub - lb <= 1); //input vector contains at most one item that is equal to the new item
```

Comment:
the docu states: "...and unique according the predicate". Then, this shouldnt happen?